mypyc/lib-rt/misc_ops.c

```diff
@@ -669,9 +669,58 @@ CPy_Super(PyObject *builtins, PyObject *self) {
     return result;
 }
 
+static bool import_single(PyObject *mod_id, PyObject **mod_static,
+                          PyObject *globals_id, PyObject *globals_name, PyObject *globals) {
+    if (*mod_static == Py_None) {
+        CPyModule *mod = PyImport_Import(mod_id);
+        if (mod == NULL) {
+            return false;
+        }
+        *mod_static = mod;
+    }
+
+    PyObject *mod_dict = PyImport_GetModuleDict();
```

Comment:
If it does, [it will raise a fatal Python error](https://github.com/python/cpython/blob/8af8f52d175959f03cf4a2786b6a81ec09e15e95/Python/import.c#L206-L208) (and presumably cause the process to exit?). In any case, [it's not documented to be able to fail](https://docs.python.org/3/c-api/import.html#c.PyImport_GetModuleDict).